node/src/Cardano/Node/API.hs

```diff
@@ -144,11 +151,32 @@ launchNodeServer
         (pure app)
         (BS8.unpack ipAddress)
         portNumber
-        (do guard (nodeBackendDebugMode params)
+        (do guard (not isDebug)
             nodeBackendTLSParams params)
-        Nothing -- TODO: Set the onException to print out JSend compliant errors
+        (Just exceptionResponse)
         Nothing -- TODO: Set a port callback for shutdown/IPC
   where
+    isDebug = nodeBackendDebugMode params
+    exceptionResponse =
+        setOnExceptionResponse handler defaultSettings
+    handler (SomeException exn)
+        | isDebug =
+            responseLBS badRequest400 [applicationJson]
+                $ encode
+                $ object
+                [ "message" .= show @Text (typeOf exn)
+                , "status" .= ErrorStatus
+                , "diagnostic" .= show @Text exn
+                ]
+        | otherwise =
+            responseLBS badRequest400 [applicationJson]
+                $ encode
+                $ object
+                [ "message" .= id @Text "UnknownError"
+                , "status" .= ErrorStatus
+                , "diagnostic" .= id @Text "An unknown error has occurred."
+                ]
```

Comment:
Any ways to leverage actual data-types here :thinking: ? 

```hs
encode $ UnknownError "An unknown error has occured"
```

I am afraid of those kind of inline encodings as they're usually source of discrepancies.

Reply:
I could define a one-off datatype with the same encoding, and later upstream it so that the wallet and the node used it.